packages/react/src/hooks/useIonModal.ts

```diff
@@ -8,14 +8,18 @@ import type { ReactComponentOrElement } from '../models/ReactComponentOrElement'
 import type { HookOverlayOptions } from './HookOverlayOptions';
 import { useOverlay } from './useOverlay';
 
-// TODO(FW-2959): types
-
 /**
  * A hook for presenting/dismissing an IonModal component
  * @param component The component that the modal will show. Can be a React Component, a functional component, or a JSX Element
  * @param componentProps The props that will be passed to the component, if required
  * @returns Returns the present and dismiss methods in an array
  */
+export function useIonModal(component: JSX.Element, componentProps?: any): UseIonModalResult;
```

Comment:
I wasn't sure on this - does it make sense to allow passing props to a `JSX.Element`? I don't think so, but I didn't want to break apps if there is some behavior I am unaware of.

If `JSX.Element` should never take `componentProps`, then I can update to remove the second parameter from this definition.